tests/unit/test_connection.py

```diff
@@ -382,8 +384,8 @@ def send_msg(msg, req_id, msg_callback):
         connection.send_msg.assert_has_calls([call(ANY, request_id, ANY)] * get_holders.call_count)
         connection.defunct.assert_has_calls([call(ANY)] * get_holders.call_count)
         exc = connection.defunct.call_args_list[0][0][0]
-        self.assertIsInstance(exc, Exception)
-        self.assertEqual(exc.args, Exception('Connection heartbeat failure').args)
+        self.assertIsInstance(exc, ConnectionException)
+        self.assertRegex(exc.args[0], r'^Received unexpected response to OptionsMessage.*')
```

Comment:
argh! I was thrown off by having unittest2 installed in my environment. Revisedin 6684e1a